.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,16 @@
+# Checklist
+
+- Branch
+    - [ ] Commit sequence broadly makes sense
+    - [ ] Commits have useful messages
+    - [ ] At every intermediate commit, the libraries must compile. It is fine if tests don't compile/pass at these commits
+    - [ ] New tests are added if needed and existing tests are updated
+    - [ ] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
```

Comment:
Could you also drop a similar line about network related changes, they should all be recorded in `../docs/interface-CHANGELOG.md`.

Reply:
Added :+1: